app/Main.hs

```diff
@@ -1,6 +1,37 @@
 module Main where
 
-import Lib
+import Control.Monad (join)
+
+import Options.Applicative
+
+import Cache (cacheBoards)
+import Search (searchBoards)
+
+data Command
+    = Search (Maybe String)
+    | Cache
 
 main :: IO ()
-main = someFunc
+main = run =<< execParser
+    (parseCommand `withInfo` "Interact with the Trello API")
+
+run :: Command -> IO ()
+run cmd =
+    case cmd of
+        Search mquery -> searchBoards mquery
+        Cache         -> cacheBoards
+
+withInfo :: Parser a -> String -> ParserInfo a
+withInfo opts desc = info (helper <*> opts) $ progDesc desc
+
+parseCommand :: Parser Command
+parseCommand =
+    subparser $
+        command "search" (parseSearch `withInfo` "Search for Trello boards") <>
```

Comment:
I'm not convinced, but I _think_ I prefer it as it is. It's likely more commands would be added in the future, and I'd rather make changes in just one place. I don't think it affects readability too much.